reflection/serverreflection.go

```diff
@@ -351,11 +311,12 @@ func (s *serverReflectionServer) fileDescEncodingContainingSymbol(name string, s
 	if fd == nil {
 		// Check if it's a type name that was not present in the
 		// transitive dependencies of the registered services.
-		if st, err := typeForName(name); err == nil {
-			fd, err = s.fileDescForType(st)
-			if err != nil {
-				return nil, err
-			}
+		desc, err := protoregistry.GlobalTypes.FindMessageByName(protoreflect.FullName(name))
+		if err != nil && err != protoregistry.NotFound {
+			panic(err)
+		}
+		if desc != nil {
```

Comment:
Is this `if` necessary even with the error checks above? I don't think `FindMessageByName` would return `nil, nil`. Or is it because of `protoregistry.NotFound`?